hal/common/retarget.cpp

```diff
@@ -713,6 +713,39 @@ extern "C" void __env_unlock( struct _reent *_r )
 {
     __rtos_env_unlock(_r);
 }
+
+extern "C" int __cxa_guard_acquire (uint64_t *guard_object_p)
```

Comment:
Good catch @pan- . It is also described on ARM pages, that they use ``int*`` as a parameter, even for ARM6 compiler.